lib/util.js

```diff
@@ -738,7 +738,9 @@ function formatSet(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   value.forEach(function(v) {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, v, nextRecurseTimes);
+    var str = ctx.seen.includes(v) ?
+      ctx.stylize('[Circular]', 'special') :
+      formatValue(ctx, v, nextRecurseTimes);
```

Comment:
One non-breaking workaround would be to continue to make `context.seen` an array and create a new `Set` internally, and use the `Set` for all membership checks. That way we would sill avoid linear-time membership checks.